rust/psibase/src/time.rs

```diff
@@ -72,32 +73,48 @@ impl<'de> Deserialize<'de> for TimePointSec {
 }
 
 #[derive(
-    Debug,
-    Copy,
-    Clone,
-    Default,
-    PartialEq,
-    Eq,
-    PartialOrd,
-    Ord,
-    Pack,
-    Unpack,
-    ToKey,
-    ToSchema,
-    SimpleObject,
-    InputObject,
+    Debug, Copy, Clone, Default, PartialEq, Eq, PartialOrd, Ord, Pack, Unpack, ToKey, ToSchema,
 )]
 #[fracpack(
     definition_will_not_change,
     fracpack_mod = "fracpack",
     custom = "TimePointUSec"
 )]
 #[to_key(psibase_mod = "crate")]
-#[graphql(input_name = "TimePointUSecInput")]
 pub struct TimePointUSec {
     pub microseconds: i64,
 }
 
+impl TryFrom<String> for TimePointUSec {
+    type Error = String;
+    fn try_from(s: String) -> Result<Self, Self::Error> {
+        DateTime::<FixedOffset>::parse_from_rfc3339(&s)
+            .map(|dt| dt.to_utc().into())
+            .map_err(|e| e.to_string())
+    }
+}
+
+impl ToString for TimePointUSec {
+    fn to_string(&self) -> String {
+        DateTime::<Utc>::from(*self).to_rfc3339()
+    }
+}
+
+#[async_graphql::Scalar]
+impl ScalarType for TimePointUSec {
```

Comment:
Does `scalar!` not work? It purports to use `Serialize` and `Deserialize` which should guarantee consistency.

Reply:
Oh, nice. Yeah that works great. Done